godot-core/src/obj/traits.rs

```diff
@@ -685,6 +685,38 @@ pub trait Singleton: GodotClass {
     fn singleton() -> Gd<Self>;
 }
 
+/// Trait for user-defined singleton classes in Godot.
+///
+/// There should be only one instance of each singleton class in the engine, accessible through [`singleton()`][Singleton::singleton].
+// For now exists mostly as a marker trait and a way to provide blanket implementation for `Singleton` trait.
+pub trait UserSingleton:
+    GodotClass + Bounds<Declarer = bounds::DeclUser, Memory = bounds::MemManual>
+{
+}
```

Comment:
Why limit to `MemManual`?

If https://github.com/godot-rust/gdext/issues/522 is the reason, should we try to fix that at some point?  🙂